src/saveSvgAsPng.js

```diff
@@ -365,15 +367,26 @@
         window.open(uri, '_temp', 'menubar=no,toolbar=no,status=no');
       }
     }
+    return;
   };
 
   out$.saveSvg = (el, name, options) => {
-    requireDomNode(el);
-    out$.svgAsDataUri(el, options || {}, uri => out$.download(name, uri));
+    return new Promise((resolve, reject) => {
+      requireDomNode(el);
```

Comment:
Can `requireDomNode(el)` be left outside the promise?

Reply:
Probably, but I think that would be a step in the wrong direction. It looks like `requireDomNode` is basically just to halt execution if `el` isn't a DOM node. If we're going to throw an error when we're returning a promise, best practice would be to call `reject` instead. Something like this:
```suggestion
return new Promise((resolve, reject) => {
  if(isDomNode(el)){
    resolve();
  } else {
    reject();
  }
});
```